### ISSUE TITLE
All correlations resulted in NaN. Please check phenotype values.

### ISSUE THREAD

**Issue by maxozo** [broadinstitute/tensorqtl#55]
Hi,
Any idea why I keep getting this issue:
`         cis_df = cis.map_cis(genotype_df, variant_df, 
    File "/usr/local/lib/python3.9/site-packages/tensorqtl/cis.py", line 589, in map_cis
      res = calculate_cis_permutations(genotypes_t, phenotype_t, permutation_ix_t,
    File "/usr/local/lib/python3.9/site-packages/tensorqtl/cis.py", line 67, in calculate_cis_permutations
      raise ValueError('All correlations resulted in NaN. Please check phenotype values.')
  ValueError: All correlations resulted in NaN. Please check phenotype values.`

It happens randomly. After trying to delve into this I think the issue is caused when all the values in permutations have equal value like - 5, 5, 5, 5, 5, 5

I have normalised my phenotypes so that gene is expressed in at least 5% of samples.
The counts ate then TMM normalised and log2 value taken. 

Thank you!

**Comment by francois-a** [comment]
Hi, what version are you using? The latest/master version should filter out phenotypes that can cause this issue. In general, it's best to filter out low-complexity phenotypes (e.g., where most values are 0, identical, etc.) beforehand. I recommend using a threshold closer to 50% since associations with phenotypes with a majority of missing values are unlikely to be meaningful.

**Comment by maxozo** [comment]
The version i am using is 1.0.5. Which is the latest one. 
Have tried now to increase the thresholdof number of samples that the gene is expressed ( (reduce number of genes tested) ) in as suggested to 50% (so at least 50% of samples has a value >0). 

I have included first 20 phenotype PCs and first 5 genotype PCs as a covariates.
Could this be causeb by low number of samples?
I am calling eqtls of different celltypes. Seems like the samples with n numbers lower than 35 fail with the above error:
`<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/mo11/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/mo11/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<meta charset=utf-8>
<meta charset=utf-8>
<meta charset=utf-8>
<meta charset=utf-8>
<meta charset=utf-8>
<meta charset=utf-8>

<body link="#0563C1" vlink="#954F72">



  | Failed (n) |   |   | Success (n)
-- | -- | -- | -- | --
B_naive | 15 |   | CD8_TEM | 42
MAIT | 15 |   | CD14_Mono | 45
CD16_Mono | 21 |   | NK | 44
CD8_Naive | 33 |   | CD4_Naive | 45
CD8_TCM | 18 |   | CD4_TCM | 45



</body>

</html>


`

Is there a suggested number of samples to be used with tensorqtl?

**Comment by lw157** [comment]
I met same situation. Is this problem solved? I used v1.0.6.

**Comment by maxozo** [comment]
I am still having this issue too.

**Comment by caragraduate** [comment]
I am also challenging the same issue.

**Comment by hrstuart** [comment]
Hi, I am also having the same issue with all values being Nan with error: 

ValueError('All correlations resulted in NaN. Please check phenotype values.') ValueError: All correlations resulted in NaN. Please check phenotype values. 

This occurs with both the nominal and empirical p-values. I was able to get the same expression data to work using fastqtl but was hoping to transition to tensorqtl. I received the same error using 1.0.7 and 1.0.5.  Thanks in advance for your help!

**Comment by francois-a** [comment]
Hi, are you able to provide a test dataset that reproduces the issue?

**Comment by Tobi1kenobi** [comment]
I am having this same issue with the latest version (1.07). Is a more stringent threshold than 50% necessary?

**Comment by inabro** [comment]
> The version i am using is 1.0.5. Which is the latest one. Have tried now to increase the thresholdof number of samples that the gene is expressed ( (reduce number of genes tested) ) in as suggested to 50% (so at least 50% of samples has a value >0).
> 
> I have included first 20 phenotype PCs and first 5 genotype PCs as a covariates. Could this be causeb by low number of samples? I am calling eqtls of different celltypes. Seems like the samples with n numbers lower than 35 fail with the above error: `
> 
> <style> </style>
>  	Failed (n)	 	 	Success (n)
> B_naive	15	 	CD8_TEM	42
> MAIT	15	 	CD14_Mono	45
> CD16_Mono	21	 	NK	44
> CD8_Naive	33	 	CD4_Naive	45
> CD8_TCM	18	 	CD4_TCM	45
> `
> 
> Is there a suggested number of samples to be used with tensorqtl?

Hi, 
did you ever figure out if this problem was actually due to the smaple number being < 40? I get a similar problem when i run tensorqtl with small datasets.

**Comment by arunhpatil** [comment]
@maxozo, @francois-a


_**TLDR: Well, when I work with RNA-Seq, I encountered this error (multiple times). Thus, I investigated the genes in question, three genes I looked up (causing this error) turned out to be non-coding RNAs (such as miRNA, lncRNA, pseudogenes). If you have similar problem, try to focus on genes of interest (broadly "protein-coding" or other types of RNA).**_ 

I have encountered similar error and came up with this solution and this is a bit tedious one but worked for me. I wouldn't suggest editing the source code just to fix the problem. However, I wonder if this is potentially a solution for this problem which is rarely encountered and not reproducible with different settings/users. I tried various combinations to give high abundant genes as input expression data. Yet, I was encountering the error. Thus, I tried to find the actuall gene that is responsible. My approach may not be right and I am open to your inputs and/or suggestions. 
`Running TensorQTL v1.0.10:`

**Here's what I did:**
1. I found the error (subject of this issue):
2. I traced the error to the line in python script (`tensor/lib/python3.10/site-packages/tensorqtl/cis.py`) and put a print statement, saved it before run:
```
if corr_t.shape[0] == 0:
        print(phenotype_t)  # this is the print statement I put
```
3. This will print the phenotype values of the gene in question. Manually traced the gene by its values and 
4. Removed the gene from my expression table with simple `grep -v "gene_inQuestion" my_expression.bed > my_expression_2.bed`. 
5. I used this new bed file and it worked. 

To circle back, I re-computed all the covariates after removing this gene and it worked too. 

I hope this is helpful. 

Thank you,
Arun.

**Comment by 20richardh** [comment]
I still get the same issue from this post on 1.0.10. 

haven't tried manually removing those rows, since I have too many cell types to process
